apps/web/middleware.ts

```diff
@@ -10,10 +10,19 @@ import {
   shareUrlRoute,
   signupRoute,
 } from "@/app/middleware/endpointValidator";
+import { getToken } from "next-auth/jwt";
 import { NextResponse } from "next/server";
 import type { NextRequest } from "next/server";
 
+import { WEBAPP_URL } from "@formbricks/lib/constants";
+
 export async function middleware(request: NextRequest) {
+  const token = await getToken({ req: request });
+  const callbackUrl = request.nextUrl.searchParams.get("callbackUrl");
+  if (token && callbackUrl) {
+    return NextResponse.redirect(WEBAPP_URL + callbackUrl);
```

Comment:
When I try this with arbitrary links, it doesn't always work. Do we need to remove the old logic in the layouts to make this work?
If so, please do 😊

Reply:
It will only work for the endpoints that the middleware is configured to listen to. Currently that's:
```
  matcher: [
    "/api/auth/callback/credentials",
    "/api/v1/users",
    "/api/(.*)/client/:path*",
    "/api/v1/js/actions",
    "/api/v1/client/storage",
    "/share/(.*)/:path",
    "/auth/login",
  ]
```

I think for now, lets keep it this way and we add when needed (bec a middleware would then run everytime that endpoint is called so will add some latency)

However, if you want other endpoints to be included, I can do so too!